src/test/ui/derived-errors/issue-31997-1.stderr

```diff
@@ -1,9 +1,15 @@
-error[E0433]: failed to resolve: use of undeclared type or module `HashMap`
-  --> $DIR/issue-31997-1.rs:20:19
+error[E0597]: `line` does not live long enough
+  --> $DIR/issue-31997-1.rs:28:25
    |
-LL |     let mut map = HashMap::new();
-   |                   ^^^^^^^ use of undeclared type or module `HashMap`
+LL |         let mut parts = line.split(" [-] ");
+   |                         ^^^^ borrowed value does not live long enough
+...
+LL |     }
+   |     - `line` dropped here while still borrowed
+LL | 
+LL |     println!("{:?}", map);
+   |                      --- borrow later used here
```

Comment:
This is not what this test is supposed to test. I suggest changing `HashMap` to `BTreeMap` instead.

Reply:
Please don't make a change like this and then undo it in the same patch series. Please rebase/squash accordingly.